Mathlib/LinearAlgebra/LinearIndependent/Lemmas.lean

```diff
@@ -248,21 +248,95 @@ lemma LinearIndepOn.pair_iff {i j : ι} (f : ι → M) (hij : i ≠ j) :
     Finset.coe_insert, Finset.coe_singleton, LinearIndepOn.pair_iff _ (by trivial)]
   simp
 
-lemma LinearIndependent.pair_symm_iff {x y : M} :
+lemma LinearIndependent.pair_symm_iff :
     LinearIndependent R ![x, y] ↔ LinearIndependent R ![y, x] := by
   suffices ∀ x y : M, LinearIndependent R ![x, y] → LinearIndependent R ![y, x] by tauto
   simp only [LinearIndependent.pair_iff]
   intro x y h s t
   specialize h t s
   rwa [add_comm, and_comm]
 
+@[simp] lemma LinearIndependent.pair_neg_left_iff :
+    LinearIndependent R ![-x, y] ↔ LinearIndependent R ![x, y] := by
+  rw [pair_iff, pair_iff]
+  refine ⟨fun h s t hst ↦ ?_, fun h s t hst ↦ ?_⟩ <;> simpa using h (-s) t (by simpa using hst)
+
+@[simp] lemma LinearIndependent.pair_neg_right_iff :
+    LinearIndependent R ![x, -y] ↔ LinearIndependent R ![x, y] := by
+  rw [pair_symm_iff, pair_neg_left_iff, pair_symm_iff]
+
+variable {S : Type*} [CommRing S] [Module S R] [Module S M]
+  [SMulCommClass S R M] [IsScalarTower S R M] [NoZeroSMulDivisors S R]
+  (a b c d : S)
+
+lemma LinearIndependent.pair_smul_iff {u : S} (hu : u ≠ 0) :
+    LinearIndependent R ![u • x, u • y] ↔ LinearIndependent R ![x, y] := by
+  simp only [LinearIndependent.pair_iff]
+  refine ⟨fun h s t hst ↦ ?_, fun h s t hst ↦ ?_⟩
+  · exact h s t (by rw [← smul_comm u s, ← smul_comm u t, ← smul_add, hst, smul_zero])
+  · specialize h (u • s) (u • t) (by rw [smul_assoc, smul_assoc, smul_comm u s, smul_comm u t, hst])
+    exact ⟨(smul_eq_zero_iff_right hu).mp h.1, (smul_eq_zero_iff_right hu).mp h.2⟩
+
+private lemma LinearIndependent.pair_add_smul_add_smul_iff_aux (h : a * d ≠ b * c)
+    (h' : LinearIndependent R ![x, y]) :
+    LinearIndependent R ![a • x + b • y, c • x + d • y] := by
+  simp only [LinearIndependent.pair_iff] at h' ⊢
+  intro s t hst
+  specialize h' (a • s + c • t) (b • s + d • t) (by simp only [← hst, smul_add, add_smul,
+    smul_assoc, smul_comm a s, smul_comm c t, smul_comm b s, smul_comm d t]; abel)
+  obtain ⟨h₁, h₂⟩ := h'
+  constructor
+  · suffices (a * d) • s = (b * c) • s by
+      by_contra hs; exact h (_root_.smul_left_injective S hs ‹_›)
+    calc (a * d) • s
+        = d • a • s := by rw [mul_comm, mul_smul]
+      _ = - (d • c • t) := by rw [eq_neg_iff_add_eq_zero, ← smul_add, h₁, smul_zero]
+      _ = (b * c) • s := ?_
+    · rw [mul_comm, mul_smul, neg_eq_iff_add_eq_zero, add_comm, smul_comm d c, ← smul_add, h₂,
+        smul_zero]
+  · suffices (a * d) • t = (b * c) • t by
+      by_contra ht; exact h (_root_.smul_left_injective S ht ‹_›)
+    calc (a * d) • t
+        = a • d • t := by rw [mul_smul]
+      _ = - (a • b • s) := by rw [eq_neg_iff_add_eq_zero, ← smul_add, add_comm, h₂, smul_zero]
+      _ = (b * c) • t := ?_
+    · rw [mul_smul, neg_eq_iff_add_eq_zero, smul_comm a b, ← smul_add, h₁, smul_zero]
+
+@[simp] lemma LinearIndependent.pair_add_smul_add_smul_iff (h : a * d ≠ b * c) :
+    LinearIndependent R ![a • x + b • y, c • x + d • y] ↔ LinearIndependent R ![x, y] := by
```

Comment:
I think you can move the side condition into the RHS (with a slightly different proof), so that `simp` can always apply this.
```suggestion
@[simp] lemma LinearIndependent.pair_add_smul_add_smul_iff :
    LinearIndependent R ![a • x + b • y, c • x + d • y] ↔ LinearIndependent R ![x, y] \and (a * d ≠ b * c) := by
```